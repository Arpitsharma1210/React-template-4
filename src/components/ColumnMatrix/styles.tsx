import styled from "styled-components";
import { colors } from "../../theme/style.palette";
import { baseFontFamily, fontSize } from "../../theme/style.typography";

export const ColumnsContainer = styled.div`
  display: flex;
  column-gap: 175px;
`;

export const Column = styled.div`
  display: flex;
  flex-direction: column;
  gap: 35px;
`;

export const IndividualField = styled.div`
  display: flex;
  flex-direction: column;
  gap: 4px;
`;

export const FieldName = styled.div`
  font-size: ${fontSize.b1};
  font-family: ${baseFontFamily};
  line-height: 28px;
`;

export const FieldValue = styled.div`
  color: ${colors.grey100};
  font-family: ${baseFontFamily};
  font-size: ${fontSize.b2};
  line-height: 20px;
`;
