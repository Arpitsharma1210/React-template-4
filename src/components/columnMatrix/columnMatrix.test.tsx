import React from 'react';
import { render, screen } from '@testing-library/react';
import ColumnMatrix from './index';

describe('ColumnMatrix Component', () => {
  it('renders column matrix with provided data', () => {
    const data = {
      Name: 'John Doe',
      Age:  30,
      Gender: 'Male',
      City: 'New York',
      Country: 'USA',
    };
    render(<ColumnMatrix data={data} />);
    expect(screen.getByText('Name')).toBeInTheDocument();
    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('Age')).toBeInTheDocument();
    expect(screen.getByText('30')).toBeInTheDocument();
    expect(screen.getByText('Gender')).toBeInTheDocument();
    expect(screen.getByText('Male')).toBeInTheDocument();
    expect(screen.getByText('City')).toBeInTheDocument();
    expect(screen.getByText('New York')).toBeInTheDocument();
    expect(screen.getByText('Country')).toBeInTheDocument();
    expect(screen.getByText('USA')).toBeInTheDocument();
  });

  it('renders column matrix with specified number of rows', () => {
    const data = {
      Field1: 'Value1',
      Field2: 'Value2',
      Field3: 'Value3',
      Field4: 'Value4',
      Field5: 'Value5',
    };
    render(<ColumnMatrix data={data} rows={2} />);
    const fields = screen.getAllByText(/Value[1-5]/);
    expect(fields).toHaveLength(5);
  });

  it('matches snapshot', () => {
    const data = {
      Name: 'John Doe',
      Age:  30,
      Gender: 'Male',
      City: 'New York',
      Country: 'USA',
    };
    const { asFragment } = render(<ColumnMatrix data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
