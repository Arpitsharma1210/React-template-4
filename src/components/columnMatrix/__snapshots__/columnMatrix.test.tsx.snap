// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnMatrix Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnyFh hGOQnk"
  >
    <div
      class="sc-gtsqUy hZKseF"
    >
      <div
        class="sc-dlniIP gMGdig"
      >
        <div
          class="sc-hKFymg vPnoS"
        >
          Name
        </div>
        <div
          class="sc-eCAqax ghLdAc"
        >
          John Doe
        </div>
      </div>
      <div
        class="sc-dlniIP gMGdig"
      >
        <div
          class="sc-hKFymg vPnoS"
        >
          Age
        </div>
        <div
          class="sc-eCAqax ghLdAc"
        >
          30
        </div>
      </div>
      <div
        class="sc-dlniIP gMGdig"
      >
        <div
          class="sc-hKFymg vPnoS"
        >
          Gender
        </div>
        <div
          class="sc-eCAqax ghLdAc"
        >
          Male
        </div>
      </div>
    </div>
    <div
      class="sc-gtsqUy hZKseF"
    >
      <div
        class="sc-dlniIP gMGdig"
      >
        <div
          class="sc-hKFymg vPnoS"
        >
          City
        </div>
        <div
          class="sc-eCAqax ghLdAc"
        >
          New York
        </div>
      </div>
      <div
        class="sc-dlniIP gMGdig"
      >
        <div
          class="sc-hKFymg vPnoS"
        >
          Country
        </div>
        <div
          class="sc-eCAqax ghLdAc"
        >
          USA
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
